libp2p/transports/quictransport.nim

```diff
@@ -192,19 +187,30 @@ method handle*(m: QuicMuxer): Future[void] {.async: (raises: []).} =
     trace "finished handling stream"
     doAssert(chann.closed, "connection not closed by handler!")
 
-  try:
-    while not m.session.atEof:
-      let stream = await m.session.getStream(Direction.In)
-      asyncSpawn handleStream(stream)
-  except QuicTransportError as exc:
-    trace "Exception in quic handler", msg = exc.msg
+  while not m.session.atEof:
+    let stream =
+      try:
+        await m.session.getStream(Direction.In)
+      except CancelledError:
+        # keep handling, until connection is closed
+        continue
+      except QuicError as exc:
+        if exc.msg == "connection closed":
```

Comment:
Probably a good idea to raise a ConnectionError in lsquic, so we don't have to rely on the error msg, but instead have a separate `except` case

Reply:
changed in downstream branch https://github.com/vacp2p/nim-libp2p/pull/1876/files#diff-e6732ffe4393737472ba25d1a6f85933cd5b5b2c85c7356a9684e05e4de8eabcR194